include/nanogui/widget.h

```diff
@@ -186,13 +186,6 @@ class NANOGUI_EXPORT Widget : public Object {
     const std::string &tooltip() const { return mTooltip; }
     void setTooltip(const std::string &tooltip) { mTooltip = tooltip; }
 
-    /// Return current font size. If not set the default of the current theme will be returned
-    int fontSize() const;
```

Comment:
Moving these to `FontWidget` introduces the only potential API breaking change.  If people have custom widgets that relied on these, they will need to inherit from `FontWidget` now instead.

Is there a way to make this clear for users who encounter this in the future?  Is there an alternative?

Reply:
I suggest keeping a single Widget class (i.e. no ``FontWidget``). In the bigger scheme of things, widgets are small enough memory-wise that we don't need to care about e.g. some extra string fields.

Reply:
That can be done.  The reason `FontWidget` became a thing was so I could force three parameters:

```cpp
FontWidget(Widget *parent, const std::string &font, bool fontDefaultIsBold)
```

1. If `font = ""`, then `setDefaultFont` takes over and sets it based on the theme. 
2. `fontDefaultIsBold` determines what to query from `Theme` (added virtual methods `defaultFont()` and `defaultBoldFont()`.
3. When `setTheme` is called, a boolean flag `mFontExplicit` signals whether or not the user explicitly requested a font e.g. for the `Label` (also is set to `true` when `setFont` is called).

So when putting this all in `Widget`, there are two reasonable options:

1. Add those parameters to the `Widget` constructor with default values (API compatibility).  The implication is that **every** widget gets a font face set (perhaps that is ideal).
2. Require that sub-classes who want font faces first set `mFont` and `mFontDefaultIsBold` in their derived constructor, and call `setDefaultFont()` afterward.

Somewhat related, do you care if I force-push here?  Since the changes are so many, I tried to make the commits mergeable rather than squash so that in the future when people are looking at the history, they don't get overwhelmed.